### PR TITLE
tp-qemu: disable_win_update.py: some updates for this case.

### PR DESCRIPTION
**Pull request by CongSmile**
1. add step to check windows service status.
2. update the way to disable windows service.

Signed-off-by: Cong Li <coli@redhat.com>

id: 1328698

### CODE REVIEW

**Comment by xutian** [comment]
looks so complex,  I think execute disable service command then reboot guest is fine.

Thanks,
Xu

**Comment by ngu-niny** [comment]
@CongSmile Should we also consider the user who still use autotest in this patch?

**Comment by CongSmile** [comment]
Deleting service works well when service is starting.

@xutian 

Code has been updated, could you please review it again?
Thanks.

File: qemu/tests/disable_win_update.py [hunk 1]
```diff
@@ -16,15 +28,24 @@ def run(test, params, env):
...
-    logging.info("Turned off windows updates service")
+    try:
+        cmd_timeout = float(params.get("cmd_timeout"))
+        query_update_service_cmd = params.get("query_update_service_cmd")
+        error_context.context("Query windows update service.", logging.info)
+        output = session.cmd_output(query_update_service_cmd,
+                                    timeout=cmd_timeout)
+        if re.search("stopped", output.lower(), re.M):
+            logging.warning("Windows update service has been deleted already.")
+            exit(0)
```
Comment (xutian) [inline qemu/tests/disable_win_update.py]: what's this ?

File: qemu/tests/disable_win_update.py [hunk 2]
```diff
@@ -16,15 +28,24 @@ def run(test, params, env):
...
-        raise exceptions.TestFail("Failed to stop Windows update service.")
-    logging.info("Stopped Windows updates services.")
-
-    disable_update_service_cmd = params.get("disable_update_service_cmd")
-    if not utils_misc.wait_for(lambda: session.cmd_status(disable_update_service_cmd,
-                               timeout=cmd_timeout) == 0, 360, 0, 5):
-        raise exceptions.TestFail("Turn off updates service failed.")
-    logging.info("Turned off windows updates service")
+    try:
+        cmd_timeout = float(params.get("cmd_timeout"))
```
Comment (xutian) [inline qemu/tests/disable_win_update.py]: default is must

File: qemu/tests/disable_win_update.py [hunk 3]
```diff
@@ -16,15 +28,23 @@ def run(test, params, env):
...
-                               timeout=cmd_timeout) == 0, 360, 0, 5):
-        raise exceptions.TestFail("Turn off updates service failed.")
-    logging.info("Turned off windows updates service")
+    try:
+        cmd_timeout = float(params.get("cmd_timeout", 180))
+        query_update_service_cmd = params.get("query_update_service_cmd")
+        error_context.context("Query windows update service.", logging.info)
+        output = session.cmd_output(query_update_service_cmd,
+                                    timeout=cmd_timeout)
+        if re.search("stopped", output.lower(), re.M):
```
Comment (xutian) [inline qemu/tests/disable_win_update.py]: no need to print such kind of info.

**Comment by CongSmile** [comment]
13:51:50 INFO | PASS smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u1.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.disable_win_update

**Comment by CongSmile** [comment]
@xutian @ngu-niny 

Could you help review this one when you are free?
Thanks.

File: qemu/tests/disable_win_update.py [hunk 4]
```diff
@@ -11,20 +23,33 @@ def run(test, params, env):
...
     :param env: Dictionary with test environment.
     """
+    def disable_win_service(session, scname):
+        """
+        :param session: VM session.
+        :param scname: Service name.
+
+        :return: return True if scname has been disabled.
+        """
+        session.sendline("sc config %s start= disabled" % scname)
```
Comment (xutian) [inline qemu/tests/disable_win_update.py]: redundant space after word 'start='

File: qemu/tests/disable_win_update.py [hunk 5]
```diff
@@ -11,20 +23,33 @@ def run(test, params, env):
...
-                               timeout=cmd_timeout) == 0, 360, 0, 5):
-        raise exceptions.TestFail("Turn off updates service failed.")
-    logging.info("Turned off windows updates service")
+    try:
+        cmd_timeout = float(params.get("cmd_timeout", 180))
+        scname = params.get("win_update_service", "WuAuServ")
+
+        error_context.context("Turned off windows updates service.",
+                              logging.info)
+        status = utils_misc.wait_for(lambda: disable_win_service(session, scname),
```
Comment (xutian) [inline qemu/tests/disable_win_update.py]: better to move try to here, above lines can't  not raise any exception.

File: qemu/tests/disable_win_update.py (hunk 4 above)
Comment (CongSmile) [inline qemu/tests/disable_win_update.py]: Usage:
start= {boot | system | auto | demand | disabled | delayed-auto}

Comment (xutian) [inline qemu/tests/disable_win_update.py]: I think usage print space before all values in '{}', it just a print format.

File: qemu/tests/disable_win_update.py [hunk 6]
```diff
@@ -11,20 +23,33 @@ def run(test, params, env):
...
+
+    error_context.context("Turned off windows updates service.",
+                          logging.info)
+    try:
+        status = utils_misc.wait_for(lambda: disable_win_service(session, scname),
+                                     timeout=cmd_timeout)
+        if not status:
+            raise exceptions.TestFail("Turn off updates service failed.")
+        session = vm.reboot(session)
+
```
Comment (xutian) [inline qemu/tests/disable_win_update.py]: redundant line

**Comment by xutian** [comment]
other parts looks ok to me.

File: qemu/tests/disable_win_update.py (hunk 4 above)
Comment (CongSmile) [inline qemu/tests/disable_win_update.py]: From help:
A space is required between an option and its value (for example, type= own. If the space is omitted, the operation will fail.

**Comment by xutian** [comment]
merged. thanks @CongSmile and @ngu-niny

**Pull request merged**